[Bug] [kompile] - Crash for mismatched Map element

K Version
---------

```
$ kompile --version
K version:    v5.3.80-1-gb1c8430118
Build date:   Mon Jun 20 17:11:12 EEST 2022
```

Description
-----------

When defining a custom Map, if the `element` attribute does not match the actual element symbol, kompile crashes.

Input Files
-----------

bug.k
```
module BUG
  imports BOOL
  imports INT
  imports STRING

  syntax MyMapKey ::= k(Int)
  syntax MyMapValue ::= v(String)

  syntax MyMap [hook(MAP.Map)]

  syntax MyMap ::= MyMap MyMap
      [ left, function, hook(MAP.concat), klabel(_MyMap_), symbol, assoc, comm
      , unit(.MyMap), element(_M|->_), index(0), format(%1%n%2)
      ]
  syntax MyMap ::= ".MyMap"
      [ function, functional, hook(MAP.unit), klabel(.MyMap), symbol
      , latex(\dotCt{MyMap})
      ]
  syntax MyMap ::= MyMapKey "|MyMap-->" MyMapValue
      [ function, functional, hook(MAP.element), klabel(_|MyMap-->_), symbol
      , latex({#1}\mapsto{#2})
      ]

  syntax priorities _|MyMap-->_ > _MyMap_ .MyMap
  syntax non-assoc _|MyMap-->_
endmodule
```

Reproduction Steps
------------------

```
$ kompile bug.k --backend haskell --debug
java.lang.AssertionError
        at org.kframework.backend.kore.ModuleToKORE.production(ModuleToKORE.java:1622)
        at org.kframework.backend.kore.ModuleToKORE.production(ModuleToKORE.java:1614)
        at org.kframework.backend.kore.ModuleToKORE.convert(ModuleToKORE.java:1785)
        at org.kframework.backend.kore.ModuleToKORE.translateSymbol(ModuleToKORE.java:400)
        at org.kframework.backend.kore.ModuleToKORE.translateSymbols(ModuleToKORE.java:375)
        at org.kframework.backend.kore.ModuleToKORE.convert(ModuleToKORE.java:206)
        at org.kframework.backend.kore.KoreBackend.getKompiledStringAndWriteSyntaxMacros(KoreBackend.java:114)
        at org.kframework.backend.kore.KoreBackend.getKompiledString(KoreBackend.java:105)
        at org.kframework.backend.kore.KoreBackend.getKompiledString(KoreBackend.java:100)
        at org.kframework.backend.haskell.HaskellBackend.accept(HaskellBackend.java:46)
        at org.kframework.kompile.KompileFrontEnd.run(KompileFrontEnd.java:88)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:57)
        at org.kframework.main.Main.runApplication(Main.java:118)
        at org.kframework.main.Main.runApplication(Main.java:108)
        at org.kframework.main.Main.main(Main.java:56)
[Error] Internal: Uncaught exception thrown of type AssertionError
(AssertionError: null)
[Warning] Compiler: Could not find main syntax module with name BUG-SYNTAX in
definition.  Use --syntax-module to specify one. Using BUG as default.
```

Expected Behavior
-----------------

Kompile should produce a reasonable error message.

--- a/k-distribution/tests/regression-new/checkWarns/missingKlabel.k
+++ b/k-distribution/tests/regression-new/checkWarns/missingKlabel.k
@@ -1,0 +1,12 @@
+module MISSINGKLABEL-SYNTAX
+endmodule
+
+module MISSINGKLABEL
+  imports MISSINGKLABEL-SYNTAX
+
+  syntax MyMap ::= MyMap MyMap
+      [ left, function, hook(MAP.concat), klabel(_MyMap_), symbol, assoc, comm
+      , unit(.MyMap), element(_M|->_), index(0), format(%1%n%2), unused
+      ]
+
+endmodule

--- a/k-distribution/tests/regression-new/checkWarns/missingKlabel.k.out
+++ b/k-distribution/tests/regression-new/checkWarns/missingKlabel.k.out
@@ -1,0 +1,1 @@
+[Error] Compiler: Expected to find exactly one production for KLabel: .MyMap found: 0

--- a/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java
+++ b/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java
@@ -1619,8 +1619,8 @@ public class ModuleToKORE {
         if (klabel.name().equals(KLabels.INJ))
             return instantiatePolySorts ? INJ_PROD.substitute(term.klabel().params()) : INJ_PROD;
         Option<scala.collection.Set<Production>> prods = module.productionsFor().get(klabel.head());
-        assert(prods.nonEmpty());
-        assert(prods.get().size() == 1);
+        if (!(prods.nonEmpty() && prods.get().size() == 1))
+            throw KEMException.compilerError("Expected to find exactly one production for KLabel: " + klabel + " found: " + prods.getOrElse(Collections::Set).size());
         return instantiatePolySorts ? prods.get().head().substitute(term.klabel().params()) : prods.get().head();
     }
 

